src/lib/modules/blockchain_process/httpProxyOverride.js

```diff
@@ -0,0 +1,123 @@
+/* global require */
+
+const http = require('http');
+const https = require('https');
+const httpProxyWsIncoming = require('http-proxy/lib/http-proxy/passes/ws-incoming');
+const common = require('http-proxy/lib/http-proxy/common');
+
+const CRLF = '\r\n';
+
+httpProxyWsIncoming.stream = (req, socket, options, head, server, cb) => {
+  const createHttpHeader = function(line, headers) {
+    return Object.keys(headers).reduce(function (head, key) {
+      const value = headers[key];
+      if (!Array.isArray(value)) {
+        head.push(`${key}: ${value}`);
+        return head;
+      }
+      for (let i = 0; i < value.length; i++) {
+        head.push(`${key}: ${value[i]}`);
+      }
+      return head;
+    }, [line])
+      .join(CRLF) + `${CRLF}${CRLF}`;
+  };
+
+  common.setupSocket(socket);
+
+  if (head && head.length) socket.unshift(head);
+
+  const protocol = common.isSSL.test(options.target.protocol) ? https : http;
+
+  const proxyReq = protocol.request(
+    common.setupOutgoing(options.ssl || {}, options, req)
+  );
+
+  // Enable developers to modify the proxyReq before headers are sent
+  if (server) {
+    server.emit('proxyReqWs', proxyReq, req, socket, options, head);
+  }
+
+  // Error Handler
+  proxyReq.on('error', onOutgoingError);
+  proxyReq.on('response', function (res) {
+    // if upgrade event isn't going to happen, close the socket
+    if (!res.upgrade) {
+      const {httpVersion, statusCode, statusMessage, headers} = res;
+      socket.write(createHttpHeader(
+        `HTTP/${httpVersion} ${statusCode} ${statusMessage}`,
+        headers
+      ));
+      res.pipe(socket);
+    }
+  });
+
+  proxyReq.on('upgrade', function(proxyRes, proxySocket, proxyHead) {
+    proxySocket.on('error', onOutgoingError);
+
+    // Allow us to listen when the websocket has completed
+    proxySocket.on('end', function () {
+      server.emit('close', proxyRes, proxySocket, proxyHead);
+    });
+
+    // The pipe below will end proxySocket if socket closes cleanly, but not
+    // if it errors (eg, vanishes from the net and starts returning
+    // EHOSTUNREACH). We need to do that explicitly.
+    socket.on('error', function () {
```

Comment:
Also here, for all the anonymous functions above, I think this would be nicer with arrow functions.